### ISSUE TITLE
SGD problem

### ISSUE THREAD

**Issue by YooshinCho** [twtygqyy/pytorch-LapSRN#12]
I tried to reproduce LAPSRN just same as paper in CVPR, so I changed optim to SGD. Then, it makes loss go inf.  So, I changed loss term divided by batchsize as paper mentioned, but still loss go inf. Do you have any idea?

**Comment by twtygqyy** [comment]
Hi @YooshinCho, if you want to use sgd as the optimizer, you need to use gradient clipping, otherwise the gradient will be too large and result in the vanishing gradient issue (the loss will go inf as you mentioned). Check out nn.utils.clip_grad_norm for  gradient clipping.

**Comment by elenacliu** [comment]
@YooshinCho I met the same problem as yours.

**Comment by YooshinCho** [comment]
I gave up reproducing with SGD optimizer...
